### PR TITLE
[Nodes Page] modularize the nodes overview page

### PR DESCRIPTION
**Pull request by philipnrmn**
⚠️  This PR depends on #1218

This PR:
 - introduces routing for the nodes page filters
 - separates the node table and node grid views into container and component

~~Testing this PR will require the `cnvs-upgrade` branch in `reactjs-components` checked out and linked locally, and use of the `REACT_JS_COMPONENTS_LOCAL=true` EV.~~

### CODE REVIEW

File: plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+class NodesTableContainer extends mixin(StoreMixin, QueryParamsMixin) {
+
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      filters: {health: 'all', name: '', service: null},
+      filteredNodes: [],
+      nodeHealthResponseReceived: false
```
Comment (rcorral) [inline plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js]: Just pointing out to rename this one too if you agree with my other comment. `receivedNodeHealthResponse`

File: plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onResetFilter() {
+    this.props.onResetFilter(...arguments);
+
+    if (this.serviceFilter !== null && this.serviceFilter.dropdown !== null) {
+      this.serviceFilter.setDropdownValue('default');
```
Comment (rcorral) [inline plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js]: We could also think of a pattern for these that get set from refs. It took me a minute to figure out where this was being set as it wasn't obvious at first.

File: plugins/nodes/src/js/pages/NodesOverview.js [hunk 3]
```diff
@@ -247,45 +244,27 @@ var NodesOverview = React.createClass({
...
+        filterInputText={this.getFilterInputText()}
+        filterButtonContent={this.getButtonContent}
+        onFilterChange={this.handleHealthFilterChange}
+        filterItemList={nodesHealth}
+        selectedFilter={healthFilter}
+        byServiceFilter={byServiceFilter}
+        handleFilterChange={this.handleByServiceFilterChange}
+        services={data.services}
+        viewTypeRadioButtons={this.getViewTypeRadioButtons(this.resetFilter)}
+        hosts={nodesList} />
```
Comment (rcorral) [inline plugins/nodes/src/js/pages/NodesOverview.js]: Would you mind alphabetizing these please :)

File: plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  filterInputText: React.PropTypes.node,
+  filterButtonContent: React.PropTypes.func,
+  onFilterChange: React.PropTypes.func,
+  filterItemList: React.PropTypes.array.isRequired,
+  selectedFilter: React.PropTypes.string,
+  byServiceFilter: React.PropTypes.string,
+  handleFilterChange: React.PropTypes.func.isRequired,
+  services: React.PropTypes.array.isRequired,
+  viewTypeRadioButtons: React.PropTypes.node.isRequired,
+  hosts: React.PropTypes.array.isRequired
```
Comment (rcorral) [inline plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js]: 🔤  please!

File: plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+const HEALTH_FILTER_BUTTONS = ['all', 'healthy', 'unhealthy'];
+
+const METHODS_TO_BIND = ['onResetFilter'];
+
+class HostsPageContent extends React.Component {
+
+  constructor() {
+    super(...arguments);
+    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate;
```
Comment (rcorral) [inline plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js]: Curious, is this line the biggest reason why we have this intermediary component?

**Comment by philipnrmn** [comment]
I'm into fixing up the grid/list filters in this PR, but just wanted to clarify that this isn't a regression - see 1.8 behaviour:

File: plugins/nodes/src/js/pages/nodes/nodes-grid/NodesGridContainer.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    super(...arguments);
+
+    this.state = {
+      filters: {health: 'all', name: '', service: null},
+      filteredNodes: [],
+      hasLoadingError: false,
+      hiddenServices: [],
+      lastMesosStateWasEmpty: true,
+      mesosStateErrorCount: 0,
+      nodeHealthResponseReceived: false,
```
Comment (philipnrmn) [inline plugins/nodes/src/js/pages/nodes/nodes-grid/NodesGridContainer.js]: great idea!

File: plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js (hunk 2 above)
Comment (philipnrmn) [inline plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js]: Yes! This also confused me when I first encountered it. I'd really like to standardise this somehow.

File: plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js (hunk 4 above)
Comment (philipnrmn) [inline plugins/nodes/src/js/pages/nodes-overview/HostsPageContent.js]: ✔️

File: plugins/nodes/src/js/pages/NodesOverview.js (hunk 3 above)
Comment (philipnrmn) [inline plugins/nodes/src/js/pages/NodesOverview.js]: ✔️

File: plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js (hunk 1 above)
Comment (philipnrmn) [inline plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js]: ✔️

File: plugins/nodes/src/js/pages/nodes/nodes-grid/NodesGridContainer.js (hunk 6 above)
Comment (philipnrmn) [inline plugins/nodes/src/js/pages/nodes/nodes-grid/NodesGridContainer.js]: ✔️

**Comment by philipnrmn** [comment]
Addressed all of @rcorral's review points.

**Comment by MatApple** [comment]
LGTM 🚢

**Pull request merged**